streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateMaintainer.java

```diff
@@ -31,7 +31,7 @@
 
     void flushState();
 
-    void close() throws IOException;
+    void close(final boolean wipeStateStore) throws IOException;
```

Comment:
We could also not wipe out the store and let the user do it manually. A manual cleanup is actually required nowadays, thus, it's actually a small side "improvement".

Note that we wipe out the whole global task dir here -- in contrast, users could do a manual per-store wipe out... But as we do the same coarse grained wipe out for all tasks and we have already a ticket for "per store cleanup" I though it would be ok for now.

Let me know what you think.

Reply:
Thanks @mjsax , this sounds perfect to me.